lib/_stream_writable.js

```diff
@@ -547,75 +524,56 @@ function afterWrite(stream, state, count, cb) {
 
 // If there's something in the buffer waiting, then process it
 function clearBuffer(stream, state) {
+  const { buffered, bufferedIndex, objectMode } = state;
+  const bufferedLength = buffered.length - bufferedIndex;
+
+  if (!bufferedLength || state.corked || state.bufferProcessing) {
```

Comment:
Perhaps?
```suggestion
  if (bufferedLength <= 0 || state.corked > 0 || state.bufferProcessing) {
```

Reply:
I prefer it the way it is? nit?

Reply:
Yeah as long as 
> Also, as the destroy() doesn't reset the bufferedIndex this may lead to buffered.length - bufferedIndex in clearBuffer being negative and therefore missing if (!bufferedLength check.

is addressed one way or another.

Reply:
I will address destroy()

Reply:
@lundibundi: How about?

BTW, thanks for reviewing this!